src/index.ts

```diff
@@ -25,6 +25,24 @@ export interface Collection<K, V> extends Map<K, V> {
 export class Collection<K, V> extends Map<K, V> {
 	public static readonly default: typeof Collection = Collection;
 
+	/**
+	 * Obtains an element if the key exists, otherwise sets and returns the value provided by the default value generator.
+	 *
+	 * @param {*} key - The key to get from if it exists or set otherwise
+	 * @param {Function} defaultValueGenerator - A function that returns the ensured value
+	 *
+	 * @returns The existing value if any, otherwise the value provided by the default value generator.
+	 *
+	 * @example
+	 * collection.ensure(guildId, () => defaultGuildConfig);
+	 */
+	public ensure(key: K, defaultValueGenerator: (key: K, collection: this) => V): V | undefined {
```

Comment:
isn't the point of this method to always return `V` instead of `undefined` compared to get?

Reply:
```suggestion
	public ensure(key: K, defaultValueGenerator: (key: K, collection: this) => V): V {
```

Reply:
I'll preface what I'm going to say with: I'm very new to typescript.

In #40 the Non-Null Assertion Operator (`!`) was used:
```ts
public ensure(key: K, getDefaultValue: () => V): V {
    if (this.has(key)) return this.get(key)!;
```
to get around this issue:
![image](https://user-images.githubusercontent.com/22574513/143322444-ee69d99c-8b85-4f25-9ff4-d67021c2aaa5.png)

However from what I read, the operator is used to inform TypeScript that you expect the value to be non-nullish.
I was under the assumption and concerned that cases such as:
```ts
const coll = new Collection([
    ['thing1', null], // here
    ['thing2', undefined], // here
]);
coll.ensure('thing1', () => 'randomThing');
coll.ensure('thing1', () => 'randomThing');
```
would lead to unsafe typing; as we are giving it a nullish value when we explicitly told TypeScript (via `!`) that the value would not be nullish.

**I would like to know what the maintainers would prefer!**

Reply:
V in this case is the value of the collection. If the collection holds null or undefined values (why would it?) those still match V but the return type should still only be V because you will never get a key that doesn't exist on the collection

Reply:
Even if you do `this.get(key)!`, it won't matter.

If `V` is `Foo | undefined`, the return for this function will be `Foo | undefined`. It doesn't matter how the types are handled inside, TypeScript only ships the function's signature.